### PR TITLE
Adding Filament runout sensor

### PR DESCRIPTION
**Pull request by GabrieleCastellani**
* [x] The mod, firmware configuration or slicer profile is in the correct category
    folder. Printable mods go to `printer_mods/`, firmware configurations
    go to `firmware_configurations/`, slicer profiles go to `slicer_profiles/`.
    Create a subfolder with your name, and place the mods in a subfolder with
    a descriptive name within that folder, e.g.: `/printer_mods/FHeilmann/flux_capacitor`
  * [x] Folder and file naming:
    * Folders and filenames shouldn't contain spaces. Only letters `a-zA-Z`, numbers `0-9`, underscores `_`, hyphens `-` and periods `.`
    * Primary color: `part_xyz.stl`
    * Accent color: `[a]_part_xyz.stl`
    * Opaque color (Blocks light): `[o]_part_xyz.stl`
    * Clear/transparent color (Allows light): `[c]_part_xyz.stl`
    * Quantity, if more than one is needed: `part_xyz_x4.stl`
  * [x] For each mod, add a small `README.md` file to its folder with a short description
    of what the mod accomplishes. This readme can be used to add pictures, give assembly
    instructions or specify a bill of materials if the mod requires additional hardware.
  * [x] The PR modifies the top-level `README.md` of the category folder adding the 
    contribution to the table. Read the top part of the file for instructions on how
    to do this. Please preserve the alphabetical ordering while adding new rows. Make sure
    to fill out the compatibility matrix to indicate which versions of the Voron printer
    the submission is compatible with.
  * [x] The mod/configuration/profile has been tested by the person submitting the mod 
    and/or other Voron users. Make sure to add information about how the mod was tested below. 
  * [x] The mod is not merely a slight modification of an official Voron part, configuration 
    or profile (i.e. an official Voron part with a few mm added or removed or a slicer profile 
    which only modifies a few values). *(When in doubt, contact one of the admins in the 
    Voron discord before submitting the PR)*
  * [x] Submitted STLs are printable without support. *(If the mod does not meet this criterion
    join the Voron discord and ask the other users for advice on how to modify the mod such 
    that it does not require supports)*
  * [x] Submitted STL files are not corrupt. *(This can be tested by opening the STL in PrusaSlicer
    and checking if mesh errors are reported.)*
  * [x] Submitted STL files are oriented and scaled properly for printing.
  * [x] Submission includes a CAD file in the form of a `.STEP` or `.SCAD` file
  * [x] Submitted firmware configs or slicer profiles contain no sensitive data (e.g. API keys).

DIY filament runout sensor that replaces standard PTFE tube holder adding a filament sensor with a ball and a microswittch

#### Which mods/configurations/profiles are added by this PR?
DIY filament runout sensor

#### How was it tested? 
On Voron 2.4 and Trident with RRF and Klipper

### CODE REVIEW

File: printer_mods/gab-3d/filament-runout-sensor/readme.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# Filament Runout Sensor
+This is DIY filament runout sensor for Voron 2.4 and Trident
+
+This replaces standard bowen_retainer.stl.
```
Comment (MapleLeafMakers) [inline printer_mods/gab-3d/filament-runout-sensor/readme.md:4]: bow*d*en_retainer.stl

**Review by MapleLeafMakers — COMMENTED** [review]
A couple of small readme issues still.  I'm also wondering if anything can be done to improve the printability of this overhang: 
image

File: printer_mods/gab-3d/filament-runout-sensor/readme.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+If you are using the tool board 1LC and connect the filament sensor to the main board
+```
+;Filament sensor
+M950 J1 C"io3.in"
+M581 T1 P1 R0 S0
+```
+## Klipper
+```
+[filament_switch_sensor runout_sensor]
```
[the remark this answers is not in the document]
Comment (GabrieleCastellani) [reply printer_mods/gab-3d/filament-runout-sensor/readme.md:53]: ok removed

**Comment by GabrieleCastellani** [comment]
I printed several times this model with a stealth burner beta and never had problems with that overhang.
I understand your fear and I'll work on a different shape

**Pull request closed without merging**